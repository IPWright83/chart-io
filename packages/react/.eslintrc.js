module.exports = {
    env: {
        browser: true,
        es2021: true,
    },
    extends: ["eslint:recommended", "plugin:react/recommended", "plugin:@typescript-eslint/recommended"],
    overrides: [
        {
            files: "*.unit.tsx?",
            rules: {
                "valid-jsdoc": "off",
            },
        },
    ],
    parser: "@typescript-eslint/parser",
    parserOptions: {
        ecmaVersion: "latest",
        sourceType: "module",
    },
    plugins: ["react", "@typescript-eslint"],
    ignorePatterns: ["jest.config.js", ".eslintrc.js", "**/jest/*.js", "testUtils/*.ts", "*.stories.tsx"],
    rules: {
        "@typescript-eslint/no-explicit-any": "off",
        "valid-jsdoc": [
            "warn",
            {
                requireReturnType: false,
                requireParamType: false,
                requireReturn: false,
            },
        ],
        "@typescript-eslint/ban-ts-comment": "warn",
    },
};
