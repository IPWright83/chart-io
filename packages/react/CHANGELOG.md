# @chart-it/react-d3

## 0.24.0

### Minor Changes

- f462652: Allow non-array inputs for Scale/Axis

## 0.23.0

### Minor Changes

- 245192e: Allow non-array inputs for Scale/Axis

## 0.22.0

### Minor Changes

- f8166d8: Added support for Legends

## 0.21.0

### Minor Changes

- 17c4895: Test

## 0.20.0

### Minor Changes

- 0862879: Test

## 0.19.0

### Minor Changes

- bb81e3a: Test

## 0.18.0

### Minor Changes

- 494a204: Test
- 5c9ca11: testing

## 0.17.0

### Minor Changes

- 744f129: Updated Chromatic settings

## 0.16.0

### Minor Changes

- feddec9: TypeScript support

### Patch Changes

- Updated dependencies [feddec9]
  - @chart-it/d3@0.2.0
  - @chart-it/detection@0.5.0
  - @chart-it/types@0.5.0

## 0.15.0

### Minor Changes

- b0d2e5e: - Exposed more options to Axis (tickValues)
  - Fixed pointer-events on Markers causing flicker
  - Introduced an `<AutoScale>` component, which is the old `<Scale>`
  - Introduced some new warnings for protection
  - Added the ability to format tooltip popup strings

## 0.14.0

### Minor Changes

- 3f81ca2: Pick stroke color from background for bars/columns
- 3f81ca2: Resolved issue with flickering tooltips due to incorrect pointer-events

## 0.13.0

### Minor Changes

- 175dd66: Pick stroke color from background for bars/columns
- 175dd66: Resolved issue with flickering tooltips due to incorrect pointer-events

## 0.12.0

### Minor Changes

- a6cc33a: Testing changesets

### Patch Changes

- Updated dependencies [a6cc33a]
  - @chart-it/detection@0.4.0

## 0.11.0

### Minor Changes

- aa0b805: Testing changesets

### Patch Changes

- Updated dependencies [aa0b805]
  - @chart-it/detection@0.3.0

## 0.10.0

### Minor Changes

- 29b2703: Fixing peer dependencies

## 0.9.0

### Minor Changes

- 3254c1e: Vite build

## 0.8.0

### Minor Changes

- bc7e7f3: Trying to fix publishing CI

### Patch Changes

- Updated dependencies [bc7e7f3]
  - @chart-it/detection@0.2.0

## 0.7.0

### Minor Changes

- 71eaab3: Tag Test 2

## 0.6.0

### Minor Changes

- 74010a0: This is a test

## 0.5.0

### Minor Changes

- 4d17cab: Added a Rect component
- 4d17cab: Allowed more marker styling

## 0.4.0

### Minor Changes

- 1c8f120: Fixes issues with linkStores in production builds

## 0.3.0

### Minor Changes

- ccedcd6: Updated Storybook docs

## 0.2.0

### Minor Changes

- 856ea4f: Improved selector performance by introducing reselect

## 0.1.33

### Patch Changes

- c2ccdd1: Test
- Updated dependencies [c2ccdd1]
  - @chart-it/detection@0.1.33
