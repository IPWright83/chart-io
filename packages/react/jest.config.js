module.exports = {
    collectCoverageFrom: ["src/**/*.{js,jsx,ts,tsx}", "!src/**/*.d.ts"],
    coveragePathIgnorePatterns: [
        "stories/*.*",
        "types/*.*",
        "index.js",
        "uuidv4.js",
        "interpolateMultiPath.js",
        "hoc/withTheme/*.*",
        "jest",
        "data",
        "lib/testUtils",
        "storybook",
    ],
    moduleFileExtensions: ["web.js", "js", "web.ts", "ts", "web.tsx", "tsx", "json", "web.jsx", "jsx", "node"],
    moduleNameMapper: {
        "^react-native$": "react-native-web",
        "\\.(css|less)$": "identity-obj-proxy",
        "^d3-array$": "<rootDir>/node_modules/d3-array/dist/d3-array.min.js",
        "^d3-axis$": "<rootDir>/node_modules/d3-axis/dist/d3-axis.min.js",
        "^d3-color$": "<rootDir>/node_modules/d3-color/dist/d3-color.min.js",
        "^d3-ease$": "<rootDir>/node_modules/d3-ease/dist/d3-ease.min.js",
        "^d3-format$": "<rootDir>/node_modules/d3-format/dist/d3-format.min.js",
        "^d3-scale$": "<rootDir>/node_modules/d3-scale/dist/d3-scale.min.js",
        "^d3-selection$": "<rootDir>/node_modules/d3-selection/dist/d3-selection.min.js",
        "^d3-shape$": "<rootDir>/node_modules/d3-shape/dist/d3-shape.min.js",
        "^d3-time-format$": "<rootDir>/node_modules/d3-time-format/dist/d3-time-format.min.js",
        "^d3-timer$": "<rootDir>/node_modules/d3-timer/dist/d3-timer.min.js",
        "^d3-transition$": "<rootDir>/node_modules/d3-transition/dist/d3-transition.min.js",
    },
    resetMocks: true,
    roots: ["<rootDir>/src"],
    setupFiles: ["react-app-polyfill/jsdom"],
    setupFilesAfterEnv: ["<rootDir>/src/jest/setupTests.js"],
    testEnvironment: "jsdom",
    testMatch: ["**/*.unit.js", "**/*.unit.jsx", "**/*.unit.ts", "**/*.unit.tsx"],
    transformIgnorePatterns: [
        "[/\\\\]node_modules[/\\\\].+\\.(js|jsx|mjs|cjs|ts|tsx)$",
        "^.+\\.module\\.(css|sass|scss)$",
    ],
    transform: {
        "^.+\\.(js|jsx|mjs|cjs|ts|tsx)$": "ts-jest",
        "^.+\\.css$": "<rootDir>/src/jest/cssTransform.js",
        "^(?!.*\\.(js|jsx|mjs|cjs|ts|tsx|css|json)$)": "<rootDir>/src/jest/fileTransform.js",
    },
    watchPlugins: ["jest-watch-typeahead/filename", "jest-watch-typeahead/testname"],
};
