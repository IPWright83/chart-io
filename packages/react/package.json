{
  "name": "@chart-it/react-d3",
  "private": false,
  "version": "0.9.5",
  "packageManager": "pnpm@7.5.2",
  "description": "chart-it is a react-d3 charting library",
  "repository": "git@github.com:IPWright83/chart-it.git",
  "author": "\"Ian Wright\" <\"ipwright83+dev@gmail.com\">",
  "main": "dist/index.umd.js",
  "module": "dist/index.es.js",
  "types": "dist/index.d.ts",
  "homepage": "http://localhost:6006/?path=/story/introduction--page",
  "scripts": {
    "test": "jest --config jest.config.js",
    "clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist",
    "lint": "eslint -c .eslintrc.json . --ext .jsx --ext .js --ext .stories.jsx",
    "storybook": "start-storybook -p 6006",
    "deploy-storybook": "storybook-to-ghpages",
    "build-storybook": "ls && build-storybook",
    "dev": "vite",
    "build": "vite build",
    "preview": "vite preview"
  },
  "np": {
    "publish": false,
    "tests": false
  },
  "keywords": [
    "d3",
    "react",
    "chart"
  ],
  "dependencies": {
    "@chart-it/detection": "workspace:*",
    "d3": "^6.2.0",
    "d3-interpolate-path": "^2.2.3",
    "lodash": "^4.17.21",
    "reselect": "^4.1.6",
    "uuid": "^8.3.2"
  },
  "devDependencies": {
    "@babel/cli": "^7.18.9",
    "@babel/plugin-transform-react-jsx": "^7.18.6",
    "@babel/preset-env": "^7.18.9",
    "@babel/preset-react": "^7.18.6",
    "@storybook/addon-actions": "6.5.12",
    "@storybook/addon-essentials": "6.5.12",
    "@storybook/addon-interactions": "6.5.12",
    "@storybook/addon-links": "6.5.12",
    "@storybook/builder-webpack5": "6.5.12",
    "@storybook/manager-webpack5": "6.5.12",
    "@storybook/node-logger": "6.5.12",
    "@storybook/preset-create-react-app": "4.1.2",
    "@storybook/react": "6.5.12",
    "@storybook/storybook-deployer": "2.8.12",
    "@storybook/testing-library": "0.0.13",
    "@testing-library/jest-dom": "^5.16.4",
    "@testing-library/react": "^13.3.0",
    "@testing-library/user-event": "^13.5.0",
    "babel-jest": "27.5.1",
    "canvas": "^2.9.3",
    "@vitejs/plugin-react": "^3.0.0",
    "chromatic": "^6.7.0",
    "crypto": "^1.0.1",
    "eslint": "^8.3.0",
    "eslint-config-react-app": "^7.0.1",
    "eslint-webpack-plugin": "3.2.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "27.5.1",
    "jest-environment-jsdom": "27.5.1",
    "jest-image-snapshot": "^5.1.0",
    "prop-types": "^15.8.1",
    "react": "^18.2.0",
    "vite": "^4.0.0",
    "react-dom": "^18.2.0",
    "react-redux": "^8.0.2",
    "react-test-renderer": "^18.2.0",
    "redux": "^4.2.0",
    "redux-devtools-extension": "^2.13.8",
    "redux-thunk": "^2.4.1"
  },
  "peerDependencies": {
    "prop-types": "^15.8.1",
    "react": "^16.8.3 | ^17 | ^18",
    "react-dom": "^16.8 | ^17 | ^18",
    "react-redux": "^7.1 | ^8",
    "redux": "^4.0.0",
    "redux-thunk": "^2.4.0"
  }
}
