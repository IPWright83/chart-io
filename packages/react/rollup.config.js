import peerDepsExternal from "rollup-plugin-peer-deps-external";
import resolve from "@rollup/plugin-node-resolve";
import commonjs from "@rollup/plugin-commonjs";
// import typescript from "rollup-plugin-typescript2";
import postcss from "rollup-plugin-postcss";
import babel from "rollup-plugin-babel";

const packageJson = require("./package.json");

const globals = {
  lodash: "lodash",
};

export default {
  input: "src/index.js",
  output: [
    {
      file: packageJson.main,
      format: "cjs",
      sourcemap: true,
      inlineDynamicImports: true,
    },
    {
      file: packageJson.module,
      format: "esm",
      sourcemap: true,
      inlineDynamicImports: true,
    },
  ],
  plugins: [
    peerDepsExternal(),
    resolve({
      extensions: [".js", ".jsx"],
    }),
    babel({
      exclude: "node_modules/**",
      presets: ["@babel/env", "@babel/preset-react"],
      runtimeHelpers: true,
    }),
    commonjs(),
    //typescript({ useTsconfigDeclarationDir: true }),
    postcss({
      extensions: [".css"],
    }),
  ],
  external: Object.keys(globals),
  output: {
    globals,
  },
};
