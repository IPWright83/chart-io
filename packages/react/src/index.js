export * from "./components";
export * from "./hoc";
export * from "./hooks";
export * from "./store";
export * from "./types";
export * from "./utils";
export * from "./constants";
export * from "./themes";
