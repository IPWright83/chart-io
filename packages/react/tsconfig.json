{
    "compilerOptions": {
        "module": "commonjs",
        "target": "es6",
        "sourceMap": true,
        "declaration": true,
        "declarationMap": true,
        "skipLibCheck": true,
        "esModuleInterop": true,
        "jsx": "react-jsx",
        "noEmit": false,
        "moduleResolution": "node",
        "isolatedModules": true,
        "composite": true,
        "allowJs": true,
        "strict": false,
        "outDir": "./dist"
    },
    "include": ["src"]
}
