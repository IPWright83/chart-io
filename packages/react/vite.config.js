import react from "@vitejs/plugin-react";
import path from "node:path";
import { defineConfig } from "vite";
import dts from "vite-plugin-dts";

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [
    react({
      // Used to enable HMR with sub-packages
      exclude: /node_modules\/(?!@chart-io\/).*/,
    }),
    dts({
      entryRoot: "src/lib",
      outDir: "dist",
      exclude: [
        "**/*.stories.ts",
        "**/*.stories.tsx",
        "**/*.stories.d.ts",
        "**/stories/**",
        "**/*.mdx",
        "**/testUtils/**",
      ],
    }),
  ],
  build: {
    sourcemap: true,
    lib: {
      ourDir: "build",
      entry: path.resolve(__dirname, "src/lib/index.ts"),
      name: "index",
      formats: ["es", "umd"],
      fileName: (format) => `index.${format}.js`,
    },
    rollupOptions: {
      external: ["react", "react-dom"],
      output: {
        globals: {
          react: "React",
          "react-dom": "ReactDOM",
        },
      },
    },
  },
});
