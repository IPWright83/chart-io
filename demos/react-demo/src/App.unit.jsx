import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

function wait(duration = 200) {
  return new Promise((resolve) => setTimeout(resolve, duration));
}

test("renders learn react link", async () => {
  const { asFragment } = render(<App />);

  await wait(500);

  expect(asFragment()).toMatchSnapshot();
});
