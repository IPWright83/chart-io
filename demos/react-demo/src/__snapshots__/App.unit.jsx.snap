// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <svg
        class="chart-svg"
        height="500"
        style="background-color: rgb(255, 255, 255);"
        width="500"
      >
        <g
          class="g-event-receiver"
        >
          <rect
            class="chart-it event-receiver"
            height="440"
            style="fill: none; pointer-events: all;"
            transform="translate(30, 30)"
            width="440"
          />
        </g>
        <defs>
          <clippath
            id="clip-path-test-chart"
          >
            <rect
              height="440"
              width="440"
              x="30"
              y="30"
            />
          </clippath>
        </defs>
        <g
          class="g-axis"
        >
          <g
            transform="translate(30, 0)"
          >
            <g
              class="g-gridlines left"
              fill="none"
              font-family="sans-serif"
              font-size="10"
              style="color: rgb(211, 211, 211); stroke-opacity: 0.8; stroke-width: 1; pointer-events: none;"
              text-anchor="end"
            >
              <g
                class="tick"
                opacity="1"
                transform="translate(0,470.5)"
              >
                <line
                  stroke="currentColor"
                  x2="440"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,426.5)"
              >
                <line
                  stroke="currentColor"
                  x2="440"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,382.5)"
              >
                <line
                  stroke="currentColor"
                  x2="440"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,338.5)"
              >
                <line
                  stroke="currentColor"
                  x2="440"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,294.5)"
              >
                <line
                  stroke="currentColor"
                  x2="440"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,250.5)"
              >
                <line
                  stroke="currentColor"
                  x2="440"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,206.5)"
              >
                <line
                  stroke="currentColor"
                  x2="440"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,162.50000000000003)"
              >
                <line
                  stroke="currentColor"
                  x2="440"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,118.49999999999999)"
              >
                <line
                  stroke="currentColor"
                  x2="440"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,74.5)"
              >
                <line
                  stroke="currentColor"
                  x2="440"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,30.5)"
              >
                <line
                  stroke="currentColor"
                  x2="440"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-3"
                />
              </g>
            </g>
            <g
              class="chart-it axis axis-left"
              fill="none"
              font-family="sans-serif"
              font-size="10"
              style="user-select: none; color: rgb(51, 51, 51); stroke-opacity: 0.8; stroke-width: 1;"
              text-anchor="end"
            >
              <path
                class="domain"
                d="M-6,470.5H0.5V30.5H-6"
                stroke="currentColor"
              />
              <g
                class="tick"
                opacity="1"
                transform="translate(0,470.5)"
              >
                <line
                  stroke="currentColor"
                  x2="-6"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-9"
                >
                  0
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,426.5)"
              >
                <line
                  stroke="currentColor"
                  x2="-6"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-9"
                >
                  1
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,382.5)"
              >
                <line
                  stroke="currentColor"
                  x2="-6"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-9"
                >
                  2
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,338.5)"
              >
                <line
                  stroke="currentColor"
                  x2="-6"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-9"
                >
                  3
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,294.5)"
              >
                <line
                  stroke="currentColor"
                  x2="-6"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-9"
                >
                  4
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,250.5)"
              >
                <line
                  stroke="currentColor"
                  x2="-6"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-9"
                >
                  5
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,206.5)"
              >
                <line
                  stroke="currentColor"
                  x2="-6"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-9"
                >
                  6
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,162.50000000000003)"
              >
                <line
                  stroke="currentColor"
                  x2="-6"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-9"
                >
                  7
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,118.49999999999999)"
              >
                <line
                  stroke="currentColor"
                  x2="-6"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-9"
                >
                  8
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,74.5)"
              >
                <line
                  stroke="currentColor"
                  x2="-6"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-9"
                >
                  9
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(0,30.5)"
              >
                <line
                  stroke="currentColor"
                  x2="-6"
                />
                <text
                  dy="0.32em"
                  fill="currentColor"
                  x="-9"
                >
                  10
                </text>
              </g>
            </g>
          </g>
        </g>
        <g
          class="g-axis"
        >
          <g
            transform="translate(0, 470)"
          >
            <g
              class="g-gridlines bottom"
              fill="none"
              font-family="sans-serif"
              font-size="10"
              style="color: rgb(211, 211, 211); stroke-opacity: 0.8; stroke-width: 1; pointer-events: none;"
              text-anchor="middle"
            >
              <g
                class="tick"
                opacity="1"
                transform="translate(30.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="-440"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(74.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="-440"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(118.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="-440"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(162.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="-440"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(206.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="-440"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(250.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="-440"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(294.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="-440"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(338.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="-440"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(382.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="-440"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(426.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="-440"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="3"
                />
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(470.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="-440"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="3"
                />
              </g>
            </g>
            <g
              class="chart-it axis axis-bottom"
              fill="none"
              font-family="sans-serif"
              font-size="10"
              style="user-select: none; color: rgb(51, 51, 51); stroke-opacity: 0.8; stroke-width: 1;"
              text-anchor="middle"
            >
              <path
                class="domain"
                d="M30.5,6V0.5H470.5V6"
                stroke="currentColor"
              />
              <g
                class="tick"
                opacity="1"
                transform="translate(30.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="6"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="9"
                >
                  0
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(74.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="6"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="9"
                >
                  1
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(118.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="6"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="9"
                >
                  2
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(162.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="6"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="9"
                >
                  3
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(206.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="6"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="9"
                >
                  4
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(250.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="6"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="9"
                >
                  5
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(294.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="6"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="9"
                >
                  6
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(338.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="6"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="9"
                >
                  7
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(382.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="6"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="9"
                >
                  8
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(426.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="6"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="9"
                >
                  9
                </text>
              </g>
              <g
                class="tick"
                opacity="1"
                transform="translate(470.5,0)"
              >
                <line
                  stroke="currentColor"
                  y2="6"
                />
                <text
                  dy="0.71em"
                  fill="currentColor"
                  y="9"
                >
                  10
                </text>
              </g>
            </g>
          </g>
        </g>
        <g
          class="g-plot scatter"
          clip-path="url(#clip-path-test-chart)"
        >
          <circle
            class="scatter-point"
            cx="250"
            cy="250"
            r="5"
            style="stroke: rgb(107, 135, 154); fill: rgba(153, 193, 220, 0.7); opacity: 0.7;"
          />
          <circle
            class="scatter-point"
            cx="470"
            cy="30"
            r="5"
            style="stroke: rgb(107, 135, 154); fill: rgba(153, 193, 220, 0.7); opacity: 0.7;"
          />
        </g>
        <g
          class="g-markers"
        />
        <g
          class="g-droplines"
        />
        <g
          class="g-tooltip-overlay"
        />
        <g
          class="g-legend-overlay"
        />
      </svg>
    </header>
  </div>
</DocumentFragment>
`;
